### ISSUE TITLE
Bug: Parser fails to parse file with decorator

### ISSUE THREAD

**Issue by dchhetri** [typescript-eslint/typescript-eslint#7046]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

parser

### Repro Code

```typescript
import { Table, Column, Model } from 'sequelize-typescript';

interface UserAttrs {
  name: string
  email: string;
  password: string
}
@Table({
  tableName: 'user',
  timestamps: true,
  underscored: true
})
class User extends Model<UserAttrs> {
  @Column
  name!: string;

  @Column
  email!: string;

  @Column
  password!: string;

}
```

### ESLint Config

```javascript
{
  "root": true,
  "parser": "@typescript-eslint/parser",
  "plugins": ["@typescript-eslint"],
  "extends": [
    "eslint:recommended",
    "plugin:@typescript-eslint/eslint-recommended",
    "plugin:@typescript-eslint/recommended"
  ]
}
```

### tsconfig

```jsonc
{
  "extends": "expo/tsconfig.base",
  "compilerOptions": {
    "strict": true,
    "experimentalDecorators": true,
    "emitDecoratorMetadata": true
  }
}
```

### Expected Result

I expected the eslint command to not to fail at the parsing stage

### Actual Result

I get the following snippet of error
```
src/server/sequelize/models/User.ts
[error] src/server/sequelize/models/User.ts: SyntaxError: This experimental syntax requires enabling one of the following parser plugin(s): "decorators", "decorators-legacy". (8:0)
[error]    1 | import { Table, Column, Model } from 'sequelize-typescript';
[error]    2 |
[error]    3 | interface UserAttrs {
[error]    4 |   name: string
[error]    5 |   email: string;
[error]    6 |   password: string
[error]    7 | }
[error]    8 | @Table({
[error]    9 |   tableName: 'user',
[error]   10 |   timestamps: true,
[error]   11 |   underscored: true
[error]   12 | })
[error]   13 | class User extends Model<UserAttrs> {
```

I'm not sure if I'm doing something incorrectly here. I wasn't able to find the plugin the error message is talking about. 

Any help would be wonderful.

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `5.59.6` |
| `@typescript-eslint/parser`            | `5.59.6` |
| `@typescript-eslint/scope-manager`     | `X.Y.Z` |
| `@typescript-eslint/typescript-estree` | `X.Y.Z` |
| `@typescript-eslint/type-utils`        | `X.Y.Z` |
| `@typescript-eslint/utils`             | `X.Y.Z` |
| `TypeScript`                           | `4.9.5` |
| `ESLint`                               | `8.41.0` |
| `node`                                 | `16.16.0` |

**Comment by Josh-Cena** [comment]
That error message comes from the Babel parser, which means our parser (and possibly the entire ESLint config) is not used at all. A repro repo is needed to help you debug but this is not going to be a parser bug.

**Comment by dchhetri** [comment]
@Josh-Cena Can I add you as a contributor to my repo so you can take a look? The rules set in the eslintrc are taking effect so I'm not sure what's going on.

**Comment by Josh-Cena** [comment]
It would be better if you can provide an isolated repro with minimal setup. The maintainers are quite busy (as for me, I have a broken computer that can only be fixed tomorrow) and don't always have time to browser through a full repo.

**Comment by bradzacher** [comment]
Yup if you delete all your code and add a simple file that has the broke code, then take your eslint config and remove lap the irrelevant bits (really just delete stuff till it stops erroring then undo that last change) then you've really created an isolated reproduction.

In doing so you'll also be able to narrow down which part of the config is causing the issue and might not even need us to help out!

I'm going to close this because it's not a bug with our tooling - but feel free to comment back if you create an isolated reproduction and we can take a look.

**Comment by dchhetri** [comment]
Thanks. Apparently, good night sleep helps. It was prettier causing the issue. I had eslint and prettier lumped into one npm script command.
